src/azure-cli/azure/cli/command_modules/servicefabric/operations/managed_clusters.py

```diff
@@ -271,3 +276,101 @@ def add_network_security_rule(cmd,
+                              source_addr_prefixes=None):
+    try:
+        cluster = client.managed_clusters.get(resource_group_name, cluster_name)
+
+        existing_nsg = find_in_collection(cluster, 'network_security_rules', 'name', name)
+        if existing_nsg is None:
+            logger.error('NSG %s does not exist.', name)
+            return None
+        
+        if protocol == 'any':
+            protocol_val = '*'
+        elif protocol is not None:
+            protocol_val = protocol
+        else:
+            protocol_val = existing_nsg.protocol
+        
+        updated_network_securityRule = NetworkSecurityRule(name=name,
+                                                        access=access if access is not None else existing_nsg.access,
+                                                        description=description if description is not None else existing_nsg.description,
+                                                        direction=direction if direction is not None else existing_nsg.direction,
+                                                        protocol=protocol_val,
+                                                        priority=priority if priority is not None else existing_nsg.priority,
+                                                        source_port_ranges=source_port_ranges if source_port_ranges is not None else existing_nsg.source_port_ranges,
+                                                        destination_port_ranges=dest_port_ranges if dest_port_ranges is not None else existing_nsg.destination_port_ranges,
+                                                        destination_address_prefixes=dest_addr_prefixes if dest_addr_prefixes is not None else existing_nsg.destination_address_prefixes,
+                                                        source_address_prefixes=source_addr_prefixes if source_addr_prefixes is not None else existing_nsg.source_address_prefixes)
+
+        if not cluster.public_ip_prefix_id:
+                cluster.public_ip_prefix_id = None
+
+        if not cluster.public_i_pv6_prefix_id:
+                cluster.public_i_pv6_prefix_id = None
+                
+        update_in_collection(cluster, 'network_security_rules', updated_network_securityRule, 'name')
+        
+        poller = client.managed_clusters.begin_create_or_update(resource_group_name, cluster_name, cluster)
+        return LongRunningOperation(cmd.cli_ctx)(poller)
+    except HttpResponseError as ex:
+        logger.error("HttpResponseError: %s", ex)
+        raise
+
+def get_network_security_rule(client,
+                            resource_group_name,
+                            cluster_name,
+                            name):
+    try:
+        cluster = client.managed_clusters.get(resource_group_name, cluster_name)
+        nsg = find_in_collection(cluster, 'network_security_rules', 'name', name)
+        
+        if nsg is None:
+            return cluster.network_security_rules
```

Comment:
probably better to throw or return None indicating the rule they are trying to get is not found